File: src/pycropml/transpiler/antlr_py/tests/examples/SimplaceComponent/SoilTemp/src/java/SoilTemp/SoiltemperatureAuxiliary.java
import  java.io.*;
import  java.util.*;
import java.time.LocalDateTime;

public class SoiltemperatureAuxiliary
{
    private Double SnowIsolationIndex;
    
    public SoiltemperatureAuxiliary() { }
    
    public SoiltemperatureAuxiliary(SoiltemperatureAuxiliary toCopy, boolean copyAll) // copy constructor 
    {
        if (copyAll)
        {
            this.SnowIsolationIndex = toCopy.getSnowIsolationIndex();
        }
    }
    public Double getSnowIsolationIndex()
    { return SnowIsolationIndex; }

    public void setSnowIsolationIndex(Double _SnowIsolationIndex)
    { this.SnowIsolationIndex= _SnowIsolationIndex; } 
    
}

File: src/pycropml/transpiler/antlr_py/tests/examples/SimplaceComponent/Simplace_SoilTemperature/src/sirius/Simplace_SoilTemperature/SoilTemperatureAuxiliary.cs

using System;
using System.Collections.Generic;
using CRA.ModelLayer.Core;
using System.Reflection;
using CRA.ModelLayer.ParametersManagement;   

namespace SiriusQualitySoilTemperature.DomainClass
{
    public class SoilTemperatureAuxiliary : ICloneable, IDomainClass
    {
        private double _SnowIsolationIndex;
        private ParametersIO _parametersIO;

        public SoilTemperatureAuxiliary()
        {
            _parametersIO = new ParametersIO(this);
        }

        public SoilTemperatureAuxiliary(SoilTemperatureAuxiliary toCopy, bool copyAll) // copy constructor 
        {
            if (copyAll)
            {
                _SnowIsolationIndex = toCopy._SnowIsolationIndex;
            }
        }

        public double SnowIsolationIndex
        {
            get { return this._SnowIsolationIndex; }
            set { this._SnowIsolationIndex= value; } 
        }

        public string Description
        {
            get { return "SoilTemperatureAuxiliary of the component";}
        }

        public string URL
        {
            get { return "http://" ;}
        }

        public virtual IDictionary<string, PropertyInfo> PropertiesDescription
        {
            get { return _parametersIO.GetCachedProperties(typeof(IDomainClass));}
        }

        public virtual Boolean ClearValues()
        {
             _SnowIsolationIndex = default(double);
            return true;
        }

        public virtual Object Clone()
        {
            IDomainClass myclass = (IDomainClass) this.MemberwiseClone();
            _parametersIO.PopulateClonedCopy(myclass);
            return myclass;
        }
    }
}